internal/expr/query.go

```diff
@@ -132,19 +160,45 @@ func (qe *QueryExpr) ScanArgs(columns []string, outputArgs []any) ([]any, error)
 			continue
 		}
 		if idx >= len(qe.outputs) {
-			return nil, fmt.Errorf("internal error: sqlair column not in outputs (%d>=%d)", idx, len(qe.outputs))
+			return nil, nil, fmt.Errorf("internal error: sqlair column not in outputs (%d>=%d)", idx, len(qe.outputs))
 		}
-		field := qe.outputs[idx]
-		outputVal, ok := typeDest[field.structType]
-		if !ok {
-			return nil, fmt.Errorf("type %q found in query but not passed to decode", field.structType.Name())
+		typeElement := qe.outputs[idx]
+		switch te := typeElement.(type) {
+		case field:
+			outputVal, ok := typeDest[te.structType]
+			if !ok {
+				return nil, nil, fmt.Errorf("type %q found in query but not passed to decode", te.structType.Name())
+			}
+			val := outputVal.FieldByIndex(te.index)
+			if !val.CanSet() {
+				return nil, nil, fmt.Errorf("internal error: cannot set field %s of struct %s", te.name, te.structType.Name())
+			}
+			ptrs = append(ptrs, val.Addr().Interface())
+		case mapKey:
+			mapDecodeInfo, ok := mapDecodeInfos[te.mapType]
+			if !ok {
+				return nil, nil, fmt.Errorf("type %q found in query but not passed to decode", te.mapType.Name())
+			}
+			// Scan in to a new variable with the type of the maps value
+			val := reflect.New(te.mapType.Elem()).Elem()
+			addr := val.Addr().Interface()
+			ptrs = append(ptrs, addr)
+			mapDecodeInfo.valuePtrs = append(mapDecodeInfo.valuePtrs, addr)
+			mapDecodeInfo.keyIndex[te.name] = len(mapDecodeInfo.valuePtrs) - 1
```

Comment:
Could we have a map of type `map[string]any`  instead? (not talking about the M type here :P ) 
i.e. instead of keeping an index to the slice `valuePtrs`, which is a slice of `Interface{}`, could we directly have the map to keep the `Interface{}`s ? Then we wouldn't need to the `valuePtrs` slice I think.

With that, we would have:
```Go
type MapDecodeInfo struct {
	m         reflect.Value
	kv        map[string]any
}
```
Then later, we could do:

```Go
mapDecodeInfo.m.SetMapIndex(reflect.ValueOf(k), reflect.ValueOf(mapDecodeInfo.kv[k])
```

Reply:
True, thats a good point, I'll have a go at implementing that.